src/misc/VarNames.jl

```diff
@@ -0,0 +1,444 @@
+The `:s => iter` specification is shorthand for `["s[$i]" for i in iter]`.
+Similarly `:s => (iter1, iter2)` is shorthand for `["s[$i,$j]" for i in iter1, j in iter2]`,
+and likewise for three and more iterables.
+
+As an alternative `"s#" => iter` is shorthand for `["s$i" for i in iter]`.
+This also works for multiple iterators in that`"s#" => (iter1, iter2)`
+is shorthand for `["s$i$j" for i in iter1, j in iter2]`.
+If you need anything else, feel free to open an issue with AbstractAlgebra.
+
+# Examples
+
+```jldoctest; setup = :(using AbstractAlgebra)
+julia> AbstractAlgebra.variable_names(:x, :y)
+2-element Vector{Symbol}:
+ :x
+ :y
+
+julia> AbstractAlgebra.variable_names(:x => (0:0, 0:1), :y => 0:1, :z)
+5-element Vector{Symbol}:
+ Symbol("x[0,0]")
+ Symbol("x[0,1]")
+ Symbol("y[0]")
+ Symbol("y[1]")
+ :z
+
+julia> AbstractAlgebra.variable_names("x#" => (0:0, 0:1), "y#" => 0:1)
+4-element Vector{Symbol}:
+ :x00
+ :x01
+ :y0
+ :y1
+
+julia> AbstractAlgebra.variable_names("x#" => 9:11)
+3-element Vector{Symbol}:
+ :x9
+ :x10
+ :x11
+
+julia> AbstractAlgebra.variable_names(["x$i$i" for i in 1:3])
+3-element Vector{Symbol}:
+ :x11
+ :x22
+ :x33
+
+julia> AbstractAlgebra.variable_names('a':'c', 'z')
+4-element Vector{Symbol}:
+ :a
+ :b
+ :c
+ :z
+
+```
+"""
+variable_names(as::VarNames...) = variable_names(as)
+# brackets = Val(false) effectively replaces `:x` and `"x"` by `"x#"`, used by macro
+variable_names(as::Tuple{Vararg{VarNames}}, brackets::Val = Val(true)) =
+    Symbol[x for a in as for x in _variable_names(a, brackets)]
+
+_variable_names(s::VarName, ::Any) = [Symbol(s)]
+_variable_names(a::AbstractArray{<:VarName}, ::Any) = Symbol.(a)
+
+_variable_names((s, axe)::Pair{<:Union{Char, Symbol}}, ::Val{true}) = Symbol.(s, '[', check_axe(axe), ']')
+_variable_names((s, axe)::Pair{<:Union{Char, Symbol}}, ::Val{false}) = check_names(Symbol.(s, axe))
+
+_variable_names((s, axes)::Pair{<:Union{Char, Symbol}, <:Tuple}, ::Val{true}) = Symbol.(s, '[', _multi_index_strings(axes), ']')
+_variable_names((s, axes)::Pair{<:Union{Char, Symbol}, <:Tuple}, ::Val{false}) = check_names(Symbol.(s, join.(Iterators.product(axes...))))
+
+_variable_names((s, axe)::Pair{<:AbstractString}, val::Val) = _variable_names(s => (axe,), val)
+function _variable_names((s, axes)::Pair{<:AbstractString, <:Tuple}, val::Val)
+    c = count("#", s)
+    req(c <= 1, """Only a single '#' allowed, but "$s" contains $c of them.
+        Please communicate your use case to the Oscar community.""")
+    return c == 0 ? _variable_names(Symbol(s) => axes, val) :
+        _check_names([Symbol(replace(s, '#' => join(i))) for i in Iterators.product(axes...)], val)
+end
+
+"""
+    check_names(names) -> names
+
+Warn, if any of the `names` is no valid Julia identifier. Return `names`.
+"""
+function check_names(names)
+    if !all(Meta.isidentifier, names)
+        badname = first(x for x in names if !Meta.isidentifier(x))
```

Comment:
Could also do this
```suggestion
    bad = findfirst(!Meta.isidentifier, names)
    if bad !== nothing
        badname = names[bad]
```

Reply:
Actually, I think both are strange – I just use my own utility function now.